spotbugs/src/main/java/edu/umd/cs/findbugs/detect/FindOverridableMethodCall.java

```diff
@@ -0,0 +1,274 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.bcel.Const;
+import org.apache.bcel.classfile.JavaClass;
+
+import edu.umd.cs.findbugs.BugAccumulator;
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.OpcodeStack;
+import edu.umd.cs.findbugs.SourceLineAnnotation;
+import edu.umd.cs.findbugs.ba.AnalysisContext;
+import edu.umd.cs.findbugs.ba.XClass;
+import edu.umd.cs.findbugs.ba.XMethod;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import edu.umd.cs.findbugs.classfile.CheckedAnalysisException;
+import edu.umd.cs.findbugs.classfile.ClassDescriptor;
+import edu.umd.cs.findbugs.util.MultiMap;
+
+public class FindOverridableMethodCall extends OpcodeStackDetector {
+
+    private static class CallerInfo {
+        XMethod method;
+        SourceLineAnnotation sourceLine;
+
+        CallerInfo(XMethod m, SourceLineAnnotation sl) {
+            method = m;
+            sourceLine = sl;
+        }
+    }
+
+    private static final Map<XMethod, CallerInfo> callerConstructors = new HashMap<>();
+    private static final Map<XMethod, CallerInfo> callerClones = new HashMap<>();
+    private static final Map<XMethod, XMethod> callsToOverridable = new HashMap<>();
+    private static final MultiMap<XMethod, XMethod> callerToCalleeMap = new MultiMap<>(ArrayList.class);
+    private static final MultiMap<XMethod, XMethod> calleeToCallerMap = new MultiMap<>(ArrayList.class);
+
+
+    private final BugAccumulator bugAccumulator;
+
+    public FindOverridableMethodCall(BugReporter bugReporter) {
+        this.bugAccumulator = new BugAccumulator(bugReporter);
+    }
+
+    @Override
+    public void visit(JavaClass obj) {
+        super.visit(obj);
+        callerConstructors.clear();
+        callerClones.clear();
+        callsToOverridable.clear();
+        callerToCalleeMap.clear();
+        calleeToCallerMap.clear();
+    }
+
+    @Override
+    public void visitAfter(JavaClass obj) {
+        bugAccumulator.reportAccumulatedBugs();
+    }
+
+    @Override
+    public void sawOpcode(int seen) {
+        if (seen == Const.INVOKEINTERFACE || seen == Const.INVOKEVIRTUAL) {
+            XMethod method = getXMethodOperand();
+            if (method == null) {
+                return;
+            }
+            OpcodeStack.Item item = stack.getStackItem(0);
+
+            if (item.getRegisterNumber() == 0 && Const.CONSTRUCTOR_NAME.equals(getMethodName())) {
+                if (checkDirectCase(method, "MC_OVERRIDABLE_METHOD_CALL_IN_CONSTRUCTOR", LOW_PRIORITY)) {
+                    checkAndRecordCallFromConstructor(getXMethod(), method);
+                }
+
+            } else if ("clone".equals(getMethodName())
+                    && (("()" + getClassDescriptor().getSignature()).equals(getMethodSig())
+                            || "()Ljava.lang.Object;".equals(getMethodSig()))
```

Comment:
`getMethodSig()` returns a slash-formatted signature:

https://github.com/spotbugs/spotbugs/blob/cf0b796f942882eafdc6f0a86c59c6aeff3e6cbb/spotbugs/src/main/java/edu/umd/cs/findbugs/visitclass/PreorderVisitor.java#L670-L674

So we need to use `Ljava/lang/Object;` instead:

```suggestion
                            || "()Ljava/lang/Object;".equals(getMethodSig()))
```